File: siebog/src/siebog/xjaf/dnarslayer/Event.java
package siebog.xjaf.dnarslayer;

import java.io.Serializable;

public class Event implements Serializable {
	private static final long serialVersionUID = 1L;

	public static enum Kind {
		ADDED, UPDATED
	}

	private final Kind kind;
	private final String statement;

	public Event(Kind kind, String statement) {
		this.kind = kind;
		this.statement = statement;
	}

	public Kind getKind() {
		return kind;
	}

	public String getStatement() {
		return statement;
	}

	@Override
	public String toString() {
		return kind + " " + statement;
	}
}

File: dnars/src/siebog/dnars/events/Event.scala
package siebog.dnars.events

import java.io.Serializable

object EventKind extends Enumeration {
	type EventKind = Value
	val ADDED, UPDATED = Value
}

import EventKind._

class Event(val kind: EventKind, val statement: String) extends Serializable {
	override def toString = kind + " " + statement
}